staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -139,6 +142,19 @@ func newStore(c *clientv3.Client, codec runtime.Codec, newFunc, newListFunc func
 	w.getCurrentStorageRV = func(ctx context.Context) (uint64, error) {
 		return storage.GetCurrentResourceVersionFromStorage(ctx, s, newListFunc, resourcePrefix, w.objectType)
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.ConsistentListFromCache) {
+		ctx := context.Background()
```

Comment:
If CheckClient spins a goroutine, you need to find a way to connect lifecycle of this goroutine with the lifecycle of caller. One way to do that would be to make the context here cancelable and connect cancel to lifecycle of store.

Reply:
Maybe i don't understand it well but based on this comment, https://github.com/kubernetes/kubernetes/pull/124612#discussion_r1590923069 my intention was that lifecycle of `CheckClient` be separate from store and only ended with timeout or nil error. Based on my knowledge store's lifecycle ends way faster than `CheckClient` and connecting these two would be either making `CheckClient` to kill itself or drag lifecycle of store to 10 minutes in worse case scenario.

Reply:
Done, i put a goroutine to sleep for 10 minutes and cancel the function this way it should be good, it controls the `CheckControl` and doesn't effect the bootstrap.